### DISCUSSION TITLE
How can I load BelongsToMany relationship data using dependent input.

### DISCUSSION THREAD

**Discussion by rmitesh** [filamentphp/filament#6218] [Help] [answered]
So, here is the scenario..

I have users which is connected with the single departments, and department have many technologies.
and user can select their favorite technologies.
```php
// Department

/* Relationships */

public function technologies(): BelongsToMany
{
    return $this->belongsToMany(Technology::class, 'department_technology', 'technology_id', 'department_id');
}

// Technology

/* Relationships */

public function users(): BelongsToMany
{
    return $this->belongsToMany(User::class, 'technology_user', 'technology_id', 'user_id');
}

public function departments(): BelongsToMany
{
    return $this->belongsToMany(Department::class, 'department_technology', 'department_id', 'technology_id');
}

// Users

/* Relationships */

public function technologies(): BelongsToMany
{
    return $this->belongsToMany(Technology::class, 'technology_user', 'user_id', 'technology_id');
}

public function department(): BelongsTo
{
    return $this->belongsTo(Department::class);
}
```
Please refer the form layout screenshot:
![image](https://user-images.githubusercontent.com/48554454/232211501-62a8addd-30fd-408e-9e94-e6084e171712.png)

Now when I create / update the user records, select first Department and then need to loading their technologies.
but it is not it will not populating selected.

I am loading technologies using `relationships()` but it display all the department technologies. I want to display department wise technologies.

I have also checked https://filamentphp.com/docs/2.x/forms/fields#populating-automatically-from-a-relationship but found the solution.

Do I missing something?

**Comment by danharrin** [comment, marked as answer]
Maybe `->relationship('technologies', 'name', fn ($get, $query) => $query->where('department_id', $get('department_id')))`

**Reply by hasan-aftab** [reply]
Thanks Mate <3

**Comment by rmitesh** [comment]
You are awesome @danharrin !
It's worked!... &#128640;🥂


```
->relationship('technologies', 'name', function ($get, $query) {
    if ( $get('department_id') ) {
        $query->withWhereHas('departments', function($record) use($get) {
            $record->where('department_id', $get('department_id'));
        });
    }
})
```
